Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean

```diff
@@ -25,6 +24,11 @@ about `colimit F` when objects and morphisms in `C` have some algebraic structur
   and `x : Fᵢ` we have `r • x = 0` implies `x = 0`, then if `r • x = 0` for `x : colimit F`, then
   `x = 0`.
 
+## Implementation details
+
+For now, we specialize our results to `C = ModuleCat R`, which is the only place they are used.
+In the future they might be generalized by assuming a `HasForget₂ C (ModuleCat R)` instance,
+plus assertions that the module structures induced by `HasForget₂` coincide.
```

Comment:
This is the major regression of this PR: we needed to specialize these results because there is no longer a good way to say for `ModuleCat` that all its homs are linear maps.